docs/api.md

```diff
@@ -0,0 +1,366 @@
+
+```JSON
+{
+  "model":"orca",
+  "created_at":"2023-08-04T19:22:45.499127Z",
+  "done":true,
+  "total_duration":5589157167,
+  "load_duration":3013701500,
+  "sample_count":114,
+  "sample_duration":81442000,
+  "prompt_eval_count":46,
+  "prompt_eval_duration":1160282000,
+  "eval_count":113,
+  "eval_duration":1325948000
+}
+```
+
+| field                | description                                |
+| -------------------- | ------------------------------------------ |
+| model                | the name of the model                      |
+| created_at           | the time the response was generated        |
+| response             | the current token                          |
+| done                 | whether the response is complete           |
+| total_duration       | total time spent generating the response   |
+| load_duration        | time spent loading the model               |
+| sample_count         | number of samples generated                |
+| sample_duration      | time spent generating samples              |
+| prompt_eval_count    | number of times the prompt was evaluated   |
+| prompt_eval_duration | time spent evaluating the prompt           |
+| eval_count           | number of times the response was evaluated |
+| eval_duration        | time spent evaluating the response         |
+
+### Example
+
+#### Request
+
+```shell
+curl --location --request POST 'http://localhost:11434/api/generate' \
+--header 'Content-Type: text/plain' \
+--data-raw '{
+    "model": "orca",
+    "prompt": "why is the sky blue"
+}'
+```
+
+#### Response
+
+```json
+{"model":"orca","created_at":"2023-08-04T19:22:44.085127Z","response":" The","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.176425Z","response":" sky","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.18883Z","response":" appears","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.200852Z","response":" blue","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.213644Z","response":" because","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.225706Z","response":" of","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:44.237686Z","response":" a","done":false}
+...
+{"model":"orca","created_at":"2023-08-04T19:22:45.487113Z","response":".","done":false}
+{"model":"orca","created_at":"2023-08-04T19:22:45.499127Z","done":true,"total_duration":5589157167,"load_duration":3013701500,"sample_count":114,"sample_duration":81442000,"prompt_eval_count":46,"prompt_eval_duration":1160282000,"eval_count":113,"eval_duration":1325948000}
+```
+
+## Create a Model
+
+**POST /api/create**
+
+### Description
+
+**Create** takes a path to a Modelfile and creates a model. The Modelfile is documented [here](./modelfile.md).
+
+### Request
+
+The **Create** endpoint takes a JSON object with the following fields:
+
+```JSON
+{
+  "name": "modelname",
+  "path": "path to Modelfile"
```

Comment:
This should mention the Modelfile _should_ be an absolute path and _must_ be accessible on the server. If the path is relative, it must be relative to where the server process started. In most cases, this isn't useful so using an absolute path is highly recommended.

Reply:
I used absolute path in my example but wasn't sure it was needed. so this makes sense. would ~/mymodelfiles/mymodel/Modelfile work? I guess its not reliable since the server may be run by another user